zerver/lib/cache.py

```diff
@@ -173,6 +174,12 @@ def func_with_caching(*args: ParamT.args, **kwargs: ParamT.kwargs) -> ReturnT:
                 return val[0]
 
             val = func(*args, **kwargs)
+            if isinstance(val, QuerySet):  # type: ignore[misc] # https://github.com/typeddjango/django-stubs/issues/704
+                logging.warning(
+                    "cache_with_key attempted to store a full QuerySet object -- flattening using list()",
+                    stack_info=True,
+                )
+                val = list(val)
```

Comment:
We should remember to turn this into an error as soon as we’re confident it doesn’t happen, because it’s equally type-unsafe to return a `List` from something declared as returning `QuerySet`.